### ISSUE TITLE
Documentation request: accessibility considerations for Wagtail sites

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#6090]
Reviewing the issues currently marked [label:Accessibility](https://github.com/wagtail/wagtail/issues?page=1&q=sort%3Aupdated-desc+label%3AAccessibility), there are now quite a few that are about the accessibility of websites built with Wagtail, rather than the accessibility of the Wagtail admin UI. Wagtail generally tries to be un-opinionated about front-end markup, but there are areas where it has a lot of opinions, and I think it would be valuable to have a one-stop-shop document highlighting "things to consider for Wagtail websites to be accessible".

I see this as a single "Accessibility considerations for Wagtail websites" page, but it could also be split into separate parts of the documentation – feedback welcome.

Here is what this page could contain:

- Images
  - alt text. Where it comes from in Wagtail (the `title` field by default). How to change that.
    - With [guidance](https://axesslab.com/alt-texts/) on when you need alt text
    - https://github.com/wagtail/rfcs/pull/51
  - Images in rich text can override the alt text. Some implementations of the image chooser may allow that too. Document this.
  - Images in rich text should be allowed to have empty alt text
  - StreamField: https://github.com/wagtail/rfcs/pull/51
- Rich text
  - `strong` vs `b`, `em` vs `i` #4665
  - Disabling heading elements
  - Empty headings, #6526
- StreamField
  - Empty headings
- TableBlock
  - #5989
  - How to make the "caption" field mandatory if appropriate
- Embeds
  - #5982 
- Links
  - Ways to prevent "read more" or "click here" links (or how to add a custom `aria-label` for those)
     - Note – some times it’s appropriate to use short link text and this should be supported too.
  - Ways to prevent `href="None"` issues when pages are unpublished or deleted

Additionally, we could also use this page to make recommendations for accessibility resources and testing tools geared towards Wagtail websites.

---

I’ll update this list as I stumble upon more things like this.

**Comment by lb-** [comment]
@thibaudcolas where does the Wagtail userbar fit in to this? as far as I can see it does not seem to be accessible in the current state (could not find an issue, happy to raise if needed).

In our docs - we could suggest a small thing like putting this at the 'top' of template's HTML code, this makes it easier to get to for editors using keyboards only.

But yeah, we might need to ensure it can be focused on and maybe add some aria attributes.

**Comment by thibaudcolas** [comment]
@lb- good question. I think the userbar would be considered to be part of the CMS admin, so it would warrant an issue to be tracked with every other one at https://github.com/wagtail/wagtail/projects/5. I haven’t reviewed it yet so yep wouldn’t be surprised if there were issues with it.

**Comment by lb-** [comment]
Done - added https://github.com/wagtail/wagtail/issues/6108

**Comment by thibaudcolas** [comment]
Boom! https://docs.wagtail.io/en/latest/advanced_topics/accessibility_considerations.html

**Comment by thibaudcolas** [comment]
I have created a project board for everything on this list that could potentially be fixed in Wagtail (they all have their own separate issues or RFCs): https://github.com/wagtail/wagtail/projects/10.
